vaadin-charts-flow-parent/vaadin-charts-flow-svg-generator/src/main/java/com/vaadin/flow/component/charts/export/NodeRunner.java

```diff
@@ -13,23 +13,64 @@
 
 package com.vaadin.flow.component.charts.export;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-import com.vaadin.flow.server.frontend.FrontendTools;
-import com.vaadin.flow.server.frontend.FrontendToolsSettings;
+import com.vaadin.flow.server.frontend.FrontendToolsLocator;
 import com.vaadin.flow.server.frontend.FrontendUtils;
 
 class NodeRunner {
 
+    private final FrontendToolsLocator frontendToolsLocator;
+
+    NodeRunner() {
+        this(new FrontendToolsLocator());
+    }
+
+    NodeRunner(FrontendToolsLocator frontendToolsLocator) {
+        this.frontendToolsLocator = frontendToolsLocator;
+    }
+
+    /**
+     * Returns the absolute file path to a Node.js executable. First tries to
+     * find a global installation, and as a fallback looks for an installation
+     * in the Vaadin home directory. Throws if no Node.js installation could be
+     * found.
+     *
+     * @return the absolute path to the Node.js executable
+     * @throws IllegalStateException
+     *             when no Node.js installation could be found
+     */
+    String findNodeExecutable() {
```

Comment:
The logic here tries to mirror the resolving logic from the previously used `FrontendTools` class: https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java#L446